include/deal.II/matrix_free/evaluation_kernels_hanging_nodes.h

```diff
@@ -431,6 +461,1418 @@ namespace internal
     }
   };
 
+  template <typename T1, VectorizationTypes VT>
+  struct Trait;
+
+  template <typename T1>
+  struct Trait<T1, VectorizationTypes::index>
+  {
+    using value_type         = typename T1::value_type;
+    using index_type         = unsigned int;
+    using interpolation_type = value_type;
+
+    template <typename T>
+    static inline const std::array<AlignedVector<interpolation_type>, 2> &
+    get_interpolation_matrix(const T &fe_eval)
+    {
+      return fe_eval.get_shape_info()
+        .data.front()
+        .subface_interpolation_matrices_scalar;
+    }
+
+    static
+#ifndef DEBUG
+      inline DEAL_II_ALWAYS_INLINE
+#endif
```

Comment:
What do you think about simplifying the call to these #ifdefs by defining (for this file only)
```c++
#ifdef DEBUG
#define DEAL_II_ALWAYS_INLINE_RELEASE
#else
#define DEAL_II_ALWAYS_INLINE_RELEASE DEAL_II_ALWAYS_INLINE
#endif
```
and then use this flag, and at the end of the relevant section call
```c++
#undef DEAL_II_ALWAYS_INLINE_RELEASE
```
I think this gives us a good intermediate state to merge for now. We can then later analyze and see if we want to make additional changes before the release.